### ISSUE TITLE
Preprocess a raw transaction so it can be signed by Neuron

### ISSUE THREAD

**Issue by Keith-CY** [Magickbase/neuron-public-issues#314]
This issue comes from signing a transaction to upgrade a contract.

---

The transaction to sign is constructed by contract developer or dapp maintainer, and it's usually a raw transaction that follows data structure of https://github.com/nervosnetwork/ckb/tree/develop/rpc#type-transaction

Here is an example
[raw-tx.json](https://github.com/Magickbase/neuron-public-issues/files/13295234/raw-tx.json)

But transaction to sign by Neuron requires extra information because Neuron may be running offline.

Here is an example
[tx-to-sign.json](https://github.com/Magickbase/neuron-public-issues/files/13295239/tx-to-sign.json)

I've used the following script to transform the raw-tx into tx-to-sign

[export-tx-to-sign.js.zip](https://github.com/Magickbase/neuron-public-issues/files/13295249/export-tx-to-sign.js.zip)

It would be greatly helpful if transform could be done on Neuron Project Page

**Comment by Danie0918** [comment]
![preprocess_a_raw_transaction_so_it_can_be_signed_by_neuron](https://github.com/user-attachments/assets/4b73bcaf-c125-4c64-b3ad-86d7e7928fda)



- Requirement:
1. Add Tools page.
2. Add Raw Transaction Conversion Tool
    - Click/drag in to upload the raw transaction file.
    - Click the Process button to check and convert the raw transaction.
    - Click Download to download the converted file.

**Comment by Kirl70** [comment]
Design Draft(2-13--2-14):
https://www.figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=40199-56347&t=jSrokM1gUEZPpNxZ-1
<img width="906" alt="image" src="https://github.com/user-attachments/assets/57338d3c-5444-4c10-abbf-e3e581daf32e">

**Comment by Keith-CY** [comment]
> Design Draft(2-13--2-14): [figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=40199-56347&t=jSrokM1gUEZPpNxZ-1](https://www.figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=40199-56347&t=jSrokM1gUEZPpNxZ-1) <img alt="image" width="672" src="https://private-user-images.githubusercontent.com/106060898/348210251-e3d25d18-04d0-4948-b98a-b1f778b7e055.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4Nzg4MDQsIm5iZiI6MTcyMDg3ODUwNCwicGF0aCI6Ii8xMDYwNjA4OTgvMzQ4MjEwMjUxLWUzZDI1ZDE4LTA0ZDAtNDk0OC1iOThhLWIxZjc3OGI3ZTA1NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxM1QxMzQ4MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNWQ2YWExZTJhN2ViOWZmZjJjZDQ0NmFiNTRhYjYyYzNiMzgwODUyN2U2MzMwYmNmODZhZDA5OTIwMmExNGY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NZMINuhRTewhF8_XQkQP92Ff1cN3tBbVOUC_OJMpm6E">

> `Please enter the raw transaction data or drag in the raw transaction json file`

Is it allowed for users to upload json file by clicking `upload` button? Generally, `click to upload` and `drag in to upload` appear together. @Danie0918

**Comment by Danie0918** [comment]
> > Design Draft(2-13--2-14): [figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=40199-56347&t=jSrokM1gUEZPpNxZ-1](https://www.figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=40199-56347&t=jSrokM1gUEZPpNxZ-1) <img alt="image" width="672" src="https://private-user-images.githubusercontent.com/106060898/348210251-e3d25d18-04d0-4948-b98a-b1f778b7e055.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4Nzg4MDQsIm5iZiI6MTcyMDg3ODUwNCwicGF0aCI6Ii8xMDYwNjA4OTgvMzQ4MjEwMjUxLWUzZDI1ZDE4LTA0ZDAtNDk0OC1iOThhLWIxZjc3OGI3ZTA1NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxM1QxMzQ4MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNWQ2YWExZTJhN2ViOWZmZjJjZDQ0NmFiNTRhYjYyYzNiMzgwODUyN2U2MzMwYmNmODZhZDA5OTIwMmExNGY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NZMINuhRTewhF8_XQkQP92Ff1cN3tBbVOUC_OJMpm6E">
> 
> > `Please enter the raw transaction data or drag in the raw transaction json file`
> 
> Is it allowed for users to upload json file by clicking `upload` button? Generally, `click to upload` and `drag in to upload` appear together. @Danie0918

Yes, `drag in to upload` is also supported, I'll adjust the text here.

**Comment by Danie0918** [comment]
> > > Design Draft(2-13--2-14): [figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=40199-56347&t=jSrokM1gUEZPpNxZ-1](https://www.figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=40199-56347&t=jSrokM1gUEZPpNxZ-1) <img alt="image" width="672" src="https://private-user-images.githubusercontent.com/106060898/348210251-e3d25d18-04d0-4948-b98a-b1f778b7e055.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4Nzg4MDQsIm5iZiI6MTcyMDg3ODUwNCwicGF0aCI6Ii8xMDYwNjA4OTgvMzQ4MjEwMjUxLWUzZDI1ZDE4LTA0ZDAtNDk0OC1iOThhLWIxZjc3OGI3ZTA1NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxM1QxMzQ4MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNWQ2YWExZTJhN2ViOWZmZjJjZDQ0NmFiNTRhYjYyYzNiMzgwODUyN2U2MzMwYmNmODZhZDA5OTIwMmExNGY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NZMINuhRTewhF8_XQkQP92Ff1cN3tBbVOUC_OJMpm6E">
> > 
> > 
> > > `Please enter the raw transaction data or drag in the raw transaction json file`
> > 
> > 
> > Is it allowed for users to upload json file by clicking `upload` button? Generally, `click to upload` and `drag in to upload` appear together. @Danie0918
> 
> Yes, `click to upload` is also supported, I'll adjust the text here.

**Comment by yanguoyu** [comment]
https://github.com/Magickbase/websites/pull/125
